code/__HELPERS/atmospherics.dm

```diff
@@ -6,3 +6,27 @@
 
 /proc/molar_cmp_equals(a,b,epsilon = MAXIMUM_ERROR_GAS_REMOVAL)
 	return (((a + epsilon) > b) && ((a - epsilon) < b))
+
+/** A simple rudimentary gasmix to information list converter. Can be used for UIs.
+  * Args: 
+  * - gasmix: [/datum/gas_mixture]
+  * - name: String used to name the list, optional.
+  * Returns: A list parsed_gasmixes with the following structure:
+  * - parsed_gasmixes - Assoc List
+  * -- Key: name			Value: String			Desc: Gasmix Name
+  * -- Key: temperature		Value: Number			Desc: Temperature in kelvins
+  * -- Key: volume 			Value: Number			Desc: Volume in liters
+  * -- Key: pressure 		Value: Number			Desc: Pressure in kPa
+  * -- Key: gases			Value: Assoc list		Desc: List of gasses in our gasmix
+  * --- Key: gas_name 		Value: Gas Mole			Desc: Gas Name - Gas Amount pair
+  * Returned list should always be filled with keys even if value are nulls.
+  */
+/proc/gas_mixture_parser(datum/gas_mixture/gasmix, name = null)
+	. = list("gases" = list(), "name" = name, "total_moles" = null, "temperature" = null, "volume"= null, "pressure"= null)
+	for(var/gas_id in gasmix?.gases)
+		.["gases"][gasmix.gases[gas_id][GAS_META][META_GAS_NAME]] = gasmix.gases[gas_id][MOLES]
+	.["total_moles"] = gasmix?.total_moles()
+	.["temperature"] = gasmix?.temperature
+	.["volume"] = gasmix?.volume
+	.["pressure"] = gasmix?.return_pressure()
+	.["ref"] = REF(gasmix)
```

Comment:
I wanted it to still return a list filled with keys even if the gasmix is null because I feel that it's safer if the JS is accessed using `data.` and not by destructuring. Would you be fine with this? I wont mind changing it if it hinders maintainability, just thought I'd make my reasoning clear.

Reply:
you could make a return list with empty informations, something like
```
if(!gasmix)
   return list("gases" = null, "temperature" = 2.7 K, etc)
```
dunno if this setup would work, but you get the gist of it.

Reply:
right, i suppose that makes sense being put on top